### ISSUE TITLE
Distortion Correction QCs

### ISSUE THREAD

**Issue by marcolandoni** [thespacedoctor/soxspipe#75]
- Number identified: 

1. the number of identified arc lines in (RAW_TYPE = FMTCHK, recipe = xsh_predict, 1 single-pinhole) frames
2. the number of identified arc lines in (RAW_TYPE = WAVE, recipe = xsh_2dmap, 1 9-pinhole arc) frames
3. the number of identified orders (RAW_TYPE = ORDERDEF, recipe = xsh_orderpos, 1 flat and 1 single-pinhole) frames

- Gaps
Largest gap (in pixels) between identified lines in FMTCHK (1-pinhole arc) and 2DMAP (9-pinhole arc) data. We also show the number of the order in which the gap appears
1.largest gap between identified lines in format check
2. largest gap between identified lines in 2dmap
3. order with largest gap in format check
4. order with largest gap in 2dmap

- fmt_resid: Residuals of the lines in FMTCHK (1-pinhole arc) data after the annealing of the physical model. 
1. minimum residual in X
2. maximum residual in X
3. rms of residuals in X
4. minimum residual in Y
5. maximum residual in Y
6. rms of residuals in Y

-  Residuals: residuals of the ORDERDEF (1-pinhole flat) data after the fitting :
1. minimum residual in X
2. maximum residual in X
3. rms of residuals in X
4. minimum residual in Y
5. maximum residual in Y
6. rms of residuals in Y

- 2d residual: Residuals of the lines in 2DMAP (9-pinhole arc) data after the annealing of the physical model. residx/y_min/max may be large due to single outliers and are therefore not scored.

1. minimum residual in X
2. maximum residual in X
3. rms of residuals in X
4. minimum residual in Y
5. maximum residual in Y
6. rms of residuals in Y

**Comment by thespacedoctor** [comment]
[UVB XShooter Page](https://www.eso.org/observing/dfo/quality/XSHOOTER/reports/HEALTH/trend_report_NLINE_UVB_HC.html)

[VIS XShooter Page](https://www.eso.org/observing/dfo/quality/XSHOOTER/reports/HEALTH/trend_report_NLINE_VIS_HC.html)

[NIR XShooter Page](https://www.eso.org/observing/dfo/quality/XSHOOTER/reports/HEALTH/trend_report_NLINE_NIR_HC.html)

**Comment by marcolandoni** [comment]
Ask Dave data needed to be retrieved in the fits header.

**Comment by thespacedoctor** [comment]
Hi @marcolandoni, I think we may waste effort re-coding this in a notebook. I'm calculating 70% of these QCs already in the main code-base. I suggest we use this Thursday's meeting to a) get the existing QC values written neatly into the product FITS headers (with an agreed naming format) and b) code the missing QCs (if we have time). All straight into branch of the code.

This can be out first test case to make sure the QC container (persistent pandas data frame) is working.

**Comment by marcolandoni** [comment]
I agree! Let's do it with the integration of Bias QC in the next iteration togheter!

**Comment by thespacedoctor** [comment]
I've now added QCs to FMTCHK (1-pinhole arc) data. Dropping the 'Gaps' QCs as I'm not sure how useful they are. But I have added combined XY residuals alongside the required single-axis residuals. I have also added a 'PLINE' QC that gives the proportion of lines detected (compared to static line list table) as I think this is more informative than just stating the number of lines detected.

[![](https://live.staticflickr.com/65535/51999351866_680c455fa2_z.png)](https://live.staticflickr.com/65535/51999351866_680c455fa2_o.png)
